epilogue-processor/src/main/java/edu/wpi/first/epilogue/processor/LoggerGenerator.java

```diff
@@ -251,26 +255,29 @@ private void writeLoggerFile(
         out.println();
       }
 
-      // Generate static final LogMetadata fields for elements with @DependsOn annotations
-      List<Element> logMetadataElements = Stream.concat(loggableFields.stream(), loggableMethods.stream())
-          .filter(element -> element.getAnnotation(DependsOn.class) != null || 
-                           element.getAnnotation(DependsOn.Container.class) != null)
-          .sorted(Comparator.comparing(e -> e.getSimpleName().toString()))
-          .collect(Collectors.toList());
+      // Generate instance LogMetadata fields for elements with @DependsOn annotations
+      List<Element> logMetadataElements =
+          Stream.concat(loggableFields.stream(), loggableMethods.stream())
+              .filter(
+                  element ->
+                      element.getAnnotation(DependsOn.class) != null
+                          || element.getAnnotation(DependsOn.Container.class) != null)
+              .sorted(Comparator.comparing(e -> e.getSimpleName().toString()))
+              .collect(Collectors.toList());
 
       if (!logMetadataElements.isEmpty()) {
         for (var element : logMetadataElements) {
-          // Generate static final LogMetadata field for caching
+          // Generate instance LogMetadata field for caching
           out.printf(
               "  // Cached LogMetadata for element %s with @DependsOn annotations%n",
               element.getSimpleName());
-          out.printf("  private static final LogMetadata %s;%n", logMetadataFieldName(element));
+          out.printf("  private final LogMetadata %s;%n", logMetadataFieldName(element));
```

Comment:
These fields could still be static, just do the initialization inline instead of in the constructor.